### PR TITLE
feat: Agena3000 integration

### PR DESCRIPTION
**Pull request by stephanegigandet**
- script to run the Agena3000 import (modeled on the Equadis script), to be scheduled in a crontab
- added Agena3000 to the list of connected databases (used in org profiles admin section to enable them, and in the product pages to display the data sources)
- also fixed a few issues related to image upload, and slightly improved the code and documentation for image upload (a lot of it can be improved, but we can do it another time)

Part of https://github.com/openfoodfacts/openfoodfacts-server/issues/6537

### CODE REVIEW

**Review by alexgarel — APPROVED** [review]
LGTM, some minor comments.

File: lib/ProductOpener/Images.pm [hunk 1]
```diff
@@ -721,252 +757,257 @@ sub process_image_upload($$$$$$$) {
...
 			}
+		}
 
-			my $lock_path = "$product_www_root/images/products/$path/$imgid.lock";
-			while ((-e $lock_path) or (-e "$product_www_root/images/products/$path/$imgid.jpg")) {
-				$imgid++;
-				$lock_path = "$product_www_root/images/products/$path/$imgid.lock";
-			}
+		my $lock_path = "$product_www_root/images/products/$path/$imgid.lock";
+		while ((-e $lock_path) or (-e "$product_www_root/images/products/$path/$imgid.jpg")) {
```
[the remark this answers is not in the document]
Comment (alexgarel) [reply lib/ProductOpener/Images.pm:789]: > can't we first search for max imgid already present
with `ls` I mean

File: lib/ProductOpener/Images.pm [hunk 2]
```diff
@@ -721,252 +757,257 @@ sub process_image_upload($$$$$$$) {
...
-		if (1 or ($file =~ /\.($extensions)$/i)) {
-			$log->debug("file type validated") if $log->is_debug();
+		# We may have a "blob" without file name and extension
+		# try to assume it is jpeg (and let ImageMagick read it anyway if it's something else)
+		# $file .= ".jpg";
 
-			if ($file =~ /\.($extensions)$/i) {
-				$extension = lc($1) ;
-			}
+		if ($file =~ /\.($extensions)$/i) {
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply lib/ProductOpener/Images.pm:765]: sure, will do

File: lib/ProductOpener/Images.pm [hunk 3]
```diff
@@ -721,252 +757,257 @@ sub process_image_upload($$$$$$$) {
...
+			$extension = lc($1);
 			$extension eq 'jpeg' and $extension = 'jpg';
-			my $filename = get_string_id_for_lang("no_language", remote_addr(). '_' . $`);
+		}
+
+		my $filename = get_string_id_for_lang("no_language", remote_addr(). '_' . $`);
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply lib/ProductOpener/Images.pm:770]: I'll add a comment, it's just a name for a temporary file, prefixed with the user ip address + a normalized version of the original file name uploaded by the user

File: lib/ProductOpener/Images.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply lib/ProductOpener/Images.pm:788]: sure we could do that

**Pull request merged**